pkg/hubble/relay/observer.go

```diff
@@ -20,17 +20,111 @@ import (
 	"time"
 
 	observerpb "github.com/cilium/cilium/api/v1/observer"
+	relaypb "github.com/cilium/cilium/api/v1/relay"
 	"github.com/cilium/cilium/pkg/hubble/relay/queue"
-	"github.com/sirupsen/logrus"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
 
+	"github.com/golang/protobuf/ptypes"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 // ensure that Server implements the observer.ObserverServer interface.
 var _ observerpb.ObserverServer = (*Server)(nil)
 
+func retrieveFlowsFromPeer(
+	ctx context.Context,
+	conn *grpc.ClientConn,
+	req *observerpb.GetFlowsRequest,
+	flows chan<- *observerpb.GetFlowsResponse,
+) error {
+	client := observerpb.NewObserverClient(conn)
+	c, err := client.GetFlows(ctx, req)
+	if err != nil {
+		return err
+	}
+	for {
+		flow, err := c.Recv()
+		switch err {
+		case io.EOF, context.Canceled:
+			return nil
+		case nil:
+			select {
+			case flows <- flow:
+			case <-ctx.Done():
+				return nil
+			}
+		default:
+			if status.Code(err) != codes.Canceled {
+				return err
+			}
+			return nil
+		}
+	}
+}
+
+func sortFlows(
```

Comment:
As this is the only place where `sortedFlows` is written to, what about changing the prototype to:
```go
func sortFlows(
	ctx context.Context,
	flows <-chan *observerpb.GetFlowsResponse,
	qlen int,
	bufferDrainTimeout time.Duration,
) chan<- *observerpb.GetFlowsResponse {...}
```
And have `sortFlows` be responsible of creating and closing the channel (this requires spawning a goroutine in the func body rather than leaving this work to the caller)? I think the code would look a little cleaner from the caller's perspective as instead of this:
```go
       sortedFlows := make(chan *observerpb.GetFlowsResponse, qlen)
        go func() {
                defer close(sortedFlows)
                // sortFlows returns either when flows is closed or ctx expires
                sortFlows(ctx, flows, qlen, s.opts.BufferDrainTimeout, sortedFlows)
        }()
```
You'd have this:
```go
sortedFlows := sortFlows(ctx, flows, qlen, s.opts.BufferDrainTimeout)
```